src/sgame/sg_utils.cpp

```diff
@@ -943,15 +943,17 @@ std::string G_EscapeServerCommandArg( Str::StringRef str )
 // Escape a command for used in server commands (sent from client to server)
 // Difference from Cmd::Escape and normal command parsing is that newlines are allowed
 // (for commands that have multi-line output)
-char *Quote( Str::StringRef str )
+// WARNING: this function can only be called BUFFERSIZE times by the same function,
+// or bad things will happen!
+const char *Quote( Str::StringRef str )
 {
-	static char buffer[ 4 ][ MAX_STRING_CHARS ];
+	int constexpr BUFFERSIZE = 256; // must be a power of 2
```

Comment:
Surely 8 or 16 is enough? Also BUFFERSIZE is not a good name as this is a number of buffers, not the size of a buffer.

Reply:
I wrote this after talking with @cu-kai on IRC, and it seems he needs to run this function in a loop with MAX_CLIENTS iterations.

> Also BUFFERSIZE

Have to admit I didn't took much time at finding a variable name. Initially I wanted to get rid of that limit entirely, perhaps using a vector, but my short tinkering didn't let me find a satisfying solution (looked uglier than current).

Tbh I wonder how much this stuff makes sense.

Reply:
You should never store the pointer returned by this in a variable. The max limit should just accommodate when you use several of them in a single line like the print(%s %s %s) things where each argument is quoted.